flax/core/frozen_dict.py

```diff
@@ -151,12 +156,18 @@ def unfreeze(x: FrozenDict[K, V]) -> Dict[K, V]:
   Makes a mutable copy of a `FrozenDict` mutable by transforming
   it into (nested) dict.
   """
-  if not isinstance(x, (FrozenDict, dict)):
+  if isinstance(x, FrozenDict):
+    # deep copy internal state of a FrozenDict
+    # the dict branch would also work here but
+    # it is much less performant.
```

Comment:
Can you please add a tiny additional clarification -- this is due to special implementation in JAX core written in C for tree_map of specifically `dict` instances, is that right?

Reply:
done